### PR TITLE
GUVNOR-3524: Do not expose unsupported features by UI (e.g. scorecards, decision trees)

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-3524

This PR sets up the ```PermissionTreeSetup``` to include the editors we need to restrict access to; and adjusts Stunner's ```NewResourceHandler``` to prevent creation of new assets if permission is denied. Plus a bunch of tests.

### CODE REVIEW

**Comment by manstis** [comment]
Part of an emsemble:

- https://github.com/AppFormer/uberfire/pull/859
- https://github.com/kiegroup/kie-wb-common/pull/1211
- https://github.com/kiegroup/drools-wb/pull/641
- https://github.com/kiegroup/kie-wb-distributions/pull/631

**Comment by romartin** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-project-client/src/main/java/org/kie/workbench/common/stunner/bpmn/project/client/handlers/BPMNDiagramNewResourceHandler.java [hunk 1]
```diff
@@ -68,6 +81,12 @@ public IsWidget getIcon() {
         return getBPMNDiagramResourceType().getIcon();
     }
 
+    @Override
+    public boolean canCreate() {
+        return authorizationManager.authorize(PERMISSION,
+                                              sessionInfo.getIdentity());
+    }
```
Comment (dgutierr) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-project-client/src/main/java/org/kie/workbench/common/stunner/bpmn/project/client/handlers/BPMNDiagramNewResourceHandler.java]: I recommend using "authorizationManager.authorize(ActivityResourceType.EDITOR, ResourceAction.READ, sessionInfo.getIdentity())" instead. This way we avoid having to figure out and build  the permission string "editor.read.?" which is a bit low level.

Comment (manstis) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-project-client/src/main/java/org/kie/workbench/common/stunner/bpmn/project/client/handlers/BPMNDiagramNewResourceHandler.java]: Thanks @dgutierr I knew there must be a nicer way, but I couldn't work out what it was. Changed.

**Comment by manstis** [comment]
Jenkins please retest this.

**Pull request merged**